File: app/src/main/java/com/dublikunt/nclient/components/classes/Size.java
package com.dublikunt.nclient.components.classes;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;

import org.jetbrains.annotations.Contract;

public class Size implements Parcelable {
    public static final Creator<Size> CREATOR = new Creator<Size>() {
        @NonNull
        @Contract("_ -> new")
        @Override
        public Size createFromParcel(Parcel in) {
            return new Size(in);
        }

        @NonNull
        @Contract(value = "_ -> new", pure = true)
        @Override
        public Size[] newArray(int size) {
            return new Size[size];
        }
    };
    private int width, height;

    public Size(int width, int height) {
        this.width = width;
        this.height = height;
    }

    protected Size(@NonNull Parcel in) {
        width = in.readInt();
        height = in.readInt();
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(@NonNull Parcel dest, int flags) {
        dest.writeInt(width);
        dest.writeInt(height);
    }

    @Override
    public String toString() {
        return "Size{" +
            "width=" + width +
            ", height=" + height +
            '}';
    }
}

File: app/src/main/kotlin/com/dublikunt/nclient/classes/Size.kt
package com.dublikunt.nclient.classes

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator

open class Size : Parcelable {
    var width: Int
    var height: Int

    constructor(width: Int, height: Int) {
        this.width = width
        this.height = height
    }

    protected constructor(`in`: Parcel) {
        width = `in`.readInt()
        height = `in`.readInt()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(width)
        dest.writeInt(height)
    }

    override fun toString(): String {
        return "Size{" +
            "width=" + width +
            ", height=" + height +
            '}'
    }

    companion object CREATOR : Creator<Size> {
        override fun createFromParcel(parcel: Parcel): Size {
            return Size(parcel)
        }

        override fun newArray(size: Int): Array<Size?> {
            return arrayOfNulls(size)
        }
    }
}
